### DISCUSSION TITLE
Delete specific rows from ErrorRecord

### DISCUSSION THREAD

**Discussion by UKHO-AH** [pester/Pester#2319] [Q&A] [answered]
Is there any way to delete specific lines from ErrorRecord please?  I only want the first line, then the lines starting with 'Expected:' (line 5) and 'But Was:' (line 6), so I'd like to delete the lines starting ErrorRecord: Expected length: * , Actual length: *, and Strings differ at index 0.

Expected strings to be the same, because TREE be set to TRUE, but they were different.
Expected length: 4
Actual length:   5
Strings differ at index 0.
Expected: ''TRUE''
But was:  ''FALSE''
           ^
Thanks

**Comment by fflaten** [comment]
Could you explain the use case? Where would you use this and how?

Are you trying to modify it before Pester prints to screen? Or are you using `PassThru` and post-processing the result object?

**Reply by UKHO-AH** [reply]
Hi, I'm trying to export the results to a SQL table so would like to select specific rows of the ErrorRecord rather than just all of it.  So  in SQL I would like to have just:

Expected strings to be the same, because TREE be set to TRUE, but they were different.
Expected: ''TRUE''
But was: ''FALSE''

rather than the full ErrorRecord:

Expected strings to be the same, because TREE be set to TRUE, but they were different.
Expected length: 4
Actual length: 5
Strings differ at index 0.
Expected: ''TRUE''
But was: ''FALSE''
^
Thanks

**Reply by fflaten** [reply]
If you use `Run.Passthru` or `-Passthru` then you can access the error-record per test directly in the returned result-object.

You'll find a `DisplayErrorMessage`-property in it that you can trim/replace to fit your requirements.

Ex (from memory so YMMW)

```ps1
$r = Invoke-Pester -Passthru
$r.Failed | Foreach-Object {
    # Each failed test
    $_.ErrorRecord.DisplayErrorMessage
}
```

**Reply by UKHO-AH** [reply]
Thanks for your reply.  Yeah, that's pretty much the way I'm doing it.  The problem I'm having is it returns the entire errorresult, and I can't find a way to just return:
Expected strings to be the same, because TREE be set to TRUE, but they were different.
Expected: ''TRUE''
But was: ''FALSE''

**Reply by fflaten** [reply]
You'll have to use string replace for now. 

Whenever #1993 is solved this might be easier unless it also includes length/index.

**Reply by UKHO-AH** [reply]
Do you have an example of using replace please? I couldn't get it working! No problem if not :)

**Reply by fflaten** [reply]
Try
`$_.ErrorRecord.DisplayErrorMessage -replace '(?ms)^Expected length:.*?(?=Expected:)'`

**Reply by UKHO-AH** [reply]
That worked a treat thanks. Any idea how I could drop the last line too (with the ^).  Thanks again

**Reply by fflaten** [reply, marked as answer]
`$_.ErrorRecord.DisplayErrorMessage -replace '(?ms)^Expected length:.*?(Expected:.*?)\s+\^', '$1'`

**Reply by UKHO-AH** [reply]
Many thanks for your help
